packages/compat/src/v1-addon.ts

```diff
@@ -132,8 +133,10 @@ export default class V1Addon {
         // our macros don't run here in stage1
         options.plugins.ast = options.plugins.ast.filter((p: any) => !isEmbroiderMacrosPlugin(p));
         if (options.plugins.ast.length > 0) {
+          const compilerChecksum = crypto.createHash('md5').update(readFileSync(options.templateCompilerPath, 'utf-8')).digest('hex');
```

Comment:
Since we pass the path into this class in both `compat-app` and `v1-addon`, can we make `NodeTemplateCompiler` do the checksum in its constructor?

Reply:
Thanks for pointing that out, looking further, there is more we can do here to tidy this up. Especially given that we already do have a nearly-what-we-need subsystem for loading/caching & generating a checksum for the template compiler https://github.com/embroider-build/embroider/blob/46fd8277a0050769a534c7762437268f026a61ba/packages/core/src/template-compiler-node.ts#L20